### PR TITLE
Owl next new rendering

### PR DESCRIPTION
**Pull request by ged-odoo**

### CODE REVIEW

**Review by sdegueldre — CHANGES_REQUESTED** [review]

File: src/app/template_helpers.ts [hunk 1]
```diff
@@ -21,7 +23,8 @@ function callSlot(
   defaultContent?: (ctx: any, node: any, key: string) => BDom
 ): BDom {
   key = key + "__slot_" + name;
-  const slots = (ctx.props && ctx.props.slots) || {};
+  const nonReactiveProps = ctx.props && ctx.props[TARGET];
+  const slots = nonReactiveProps ? nonReactiveProps.slots || {} : {};
```
Comment (ged-odoo) [inline src/app/template_helpers.ts:27]: should markraw at the source instead

File: src/component/component.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ export class Component<Props = any, Env = any> {
 
   setup() {}
 
-  render() {
-    this.__owl__.render();
+  render(force: boolean = false) {
```
Comment (ged-odoo) [inline src/component/component.ts:39]: rename to deep

File: src/component/component_node.ts [hunk 3]
```diff
@@ -246,6 +267,9 @@ export class ComponentNode<P = any, E = any> implements VNode<ComponentNode<P, E
     this.fiber = fiber;
     const component = this.component;
     applyDefaultProps(props, component.constructor as any);
+
+    currentNode = this;
+    props = useState(props);
```
[the remark this answers is not in the document]
Comment (ged-odoo) [reply src/component/component_node.ts:272]: or somewher else

[the remark this answers is not in the document]
Comment (ged-odoo) [reply src/component/component_node.ts:272]: check if we should clearreactivecallbacks for old props

File: src/component/component_node.ts [hunk 4]
```diff
@@ -310,6 +334,14 @@ export class ComponentNode<P = any, E = any> implements VNode<ComponentNode<P, E
...
 
   patch() {
+    if (this.fiber && this.fiber.parent) {
+      // we only patch here renderings coming from above. renderings initiated
+      // by the component will be patched independently in the appropriate
+      // fiber.complete
+      this._patch();
+    }
+  }
+  _patch() {
```
Comment (ged-odoo) [inline src/component/component_node.ts:344]: maybe remove _patch

File: tests/components/basics.test.ts [hunk 5]
```diff
@@ -121,7 +121,7 @@ describe("basics", () => {
     class Test extends Component {
       static template = xml`<span>simple vnode</span>`;
       setup() {
-        expect(this.props).toBe(p);
+        expect(this.props).not.toBe(p);
```
Comment (ged-odoo) [inline tests/components/basics.test.ts:124]: toRaw

File: src/compiler/code_generator.ts [hunk 6]
```diff
@@ -1125,7 +1125,8 @@ export class CodeGenerator {
     }
 
     if (slotDef && !(ast.dynamicProps || hasSlotsProp)) {
-      props.push(`slots: ${slotDef}`);
```
Comment (ged-odoo) [inline src/compiler/code_generator.ts:1128]: remove target toraw from callslot helper

**Review by sdegueldre — APPROVED** [review]

**Pull request merged**